core/src/main/scala/org/http4s/Message.scala

```diff
@@ -123,6 +123,9 @@ sealed trait Message[F[_]] extends MessageOps[F] { self =>
       implicit F: FlatMap[F],
       decoder: EntityDecoder[F, T]): DecodeResult[F, T] =
     decoder.decode(this, strict = false)
+
+  override def decodeJson[T](implicit jsonDecoder: JsonDecoder[F, T], F: FlatMap[F]): F[T] =
```

Comment:
I imagined syntax being provided by the JSON support modules.  For example, on `CirceInstances`

```scala
implicit class MessageSyntax(self: Message[F]) {
  def decodeJson[A](implicit dec: Decoder[A], F: FlatMap[F]): F[A]
}
```

Reply:
Yea, that makes a lot more sense. It was also the original idea I had, but I got sidetracked somehow.

This still requires `msg.decodeJson(decoder, implicitly)`, but I guess we need to do that.

Reply:
If the constraint on `F` is moved to the class, you can skip the implicitly. Will finish this up and create a new PR.

Reply:
:+1: